Fix: Events API null total

tuminzee opened the pull request:
Fixes https://github.com/FrigadeHQ/trench/issues/13
Earlier the total was returning a hardcoded `null` value, I changed it to get count on the where clause

``` ts
const total = await this.clickhouse.queryResults(totalQuery, workspace.databaseName)
const totalCount = total?.[0]?.count || null
```

tuminzee commented:
@christianmat I am not sure if I am supposed to write tests for it. Anyways do let me know what you think about it.

christianmat commented:
Hey @tuminzee, thanks for this PR. Please see the guide on contributing re: tests: https://docs.trench.dev/contributing

One thing we need to fix here is that the `total` is not meant to simply equal the length of the array, it's supposed to be the total amount of records matching the query string. It is meant to be used for pagination so you know how many results you have. The real solution should be to do a separate query to clickhouse in parallel that counts the number of rows that matches the query.

tuminzee commented:
@christianmat I have a made a separate query for calculating the total. It is not a length fn.

I will check the contribution guidelines

christianmat commented on apps/trench/src/events/events.dao.ts:
```diff
@@ -103,16 +103,19 @@ export class EventsDao {
+    const totalQuery = `SELECT COUNT(*) AS count FROM events ${whereClause}`
 
     try {
       const result = await this.clickhouse.queryResults(clickhouseQuery, workspace.databaseName)
       const results = result.map((row: any) => mapRowToEvent(row))
+      const total = await this.clickhouse.queryResults(totalQuery, workspace.databaseName)
+      const totalCount = total?.[0]?.count || null
```
Can you change this to be `Promise.all` such that it executes in parallel?

christianmat reviewed (changes requested):
Please add e2e tests as well that test that this field works. See other e2e tests for examples.

tuminzee commented:
> Please add e2e tests as well that test that this field works. See other e2e tests for examples.

Yes I will do that

tuminzee commented:
[`waitForQueryResults`](https://github.com/FrigadeHQ/trench/blob/2aaa3e784e1fd869664033b2f5793e9928ce4548/apps/trench/test/e2e/utils.ts#L48) function should return the complete `res.body` instead of just `res.body.results` since we need to test other response properties like [`total`](https://github.com/FrigadeHQ/trench/blob/2aaa3e784e1fd869664033b2f5793e9928ce4548/apps/trench/src/events/events.dao.ts#L115C9-L115C14)


Questions for review:
1. Should I update all affected tests to include `total` property validation, or should we keep this as a separate enhancement?
2. If no results are found the query would always return a [timeout error](https://github.com/FrigadeHQ/trench/blob/2aaa3e784e1fd869664033b2f5793e9928ce4548/apps/trench/src/events/events.dao.ts#L115C9-L115C14), shouldn't it just return a plain response?

christianmat commented:
> [`waitForQueryResults`](https://github.com/FrigadeHQ/trench/blob/2aaa3e784e1fd869664033b2f5793e9928ce4548/apps/trench/test/e2e/utils.ts#L48) function should return the complete `res.body` instead of just `res.body.results` since we need to test other response properties like [`total`](https://github.com/FrigadeHQ/trench/blob/2aaa3e784e1fd869664033b2f5793e9928ce4548/apps/trench/src/events/events.dao.ts#L115C9-L115C14)
> 
> Questions for review:
> 
> 1. Should I update all affected tests to include `total` property validation, or should we keep this as a separate enhancement?
> 2. If no results are found the query would always return a [timeout error](https://github.com/FrigadeHQ/trench/blob/2aaa3e784e1fd869664033b2f5793e9928ce4548/apps/trench/src/events/events.dao.ts#L115C9-L115C14), shouldn't it just return a plain response?

1. Yes, please update all the existing tests to account for the new `total` behavior.
2. Not sure what you mean. It currently returns an empty array if no results are returned, it doesn't throw an error (see test screenshot below)
![image](https://github.com/user-attachments/assets/77e8527e-a8a0-4e80-8bbe-8f36c64909cd)

tuminzee commented:
- I have updated the tests to include the [total](https://github.com/tuminzee/trench/blob/754ec18743c124f10178d50a2a0834ab55035cba/apps/trench/test/e2e/utils.ts#L55-L57) property.
- Regarding the timeout error: I apologize for not clarifying earlier—this issue occurs only in the e2e tests. [waitForQueryResults](https://github.com/tuminzee/trench/blob/754ec18743c124f10178d50a2a0834ab55035cba/apps/trench/test/e2e/utils.ts#L48-L63): It currently throws an error if no results are found. Should this be the expected behavior, or should we modify it to return an empty array instead? Let me know your preference, and I can adjust accordingly. I have added a test to demonstrate this [here](https://github.com/tuminzee/trench/blob/c798969220835ab7d7e085ac76c92618c02a9c83/apps/trench/test/e2e/events.e2e.test.ts#L91-L95)
